packages/components/tests/dummy/app/templates/components/dialog.hbs

```diff
@@ -0,0 +1,172 @@
+    Here is the API for the component:
+  </p>
+  <dl class="dummy-component-props" aria-labelledby="component-api-dialog">
+
+    {{! ADD YOUR CONTENT HERE }}
+
+  </dl>
+</section>
+
+<section>
+  <h3 class="dummy-h3" id="how-to-use"><a href="#how-to-use" class="dummy-link-section">§</a> How to use</h3>
+
+  {{! ADD YOUR CONTENT HERE }}
+
+</section>
+
+<section>
+  <h3 class="dummy-h3" id="design-guidelines"><a href="#design-guidelines" class="dummy-link-section">§</a>
+    Design guidelines</h3>
+  {{! UNCOMMENT THIS BLOCK (once the link and/or the image are available) }}
+  {{!
+  <div class="dummy-design-guidelines">
+    <p class="dummy-paragraph">
+      <a href="[ADD THE LINK TO THE FIGMA FILE/PAGE HERE!]" target="_blank" rel="noopener noreferrer">Figma UI Kit</a>
+    </p>
+    <br />
+    <img class="dummy-figma-docs" src="/assets/images/dialog-design-usage.png" alt="" role="none" />
+  </div>
+  }}
+</section>
+
+<section>
+  <h3 class="dummy-h3" id="accessibility"><a href="#accessibility" class="dummy-link-section">§</a> Accessibility</h3>
+
+  {{! ADD YOUR CONTENT HERE }}
+
+</section>
+
+<section data-test-percy>
+  <h3 class="dummy-h3" id="showcase"><a href="#showcase" class="dummy-link-section">§</a> Showcase</h3>
+
+  <h4 class="dummy-h4">Color</h4>
+  {{#each @model.COLORS as |color|}}
+    <p class="dummy-paragraph">{{capitalize color}}</p>
+    <br />
+    <div class="dummy-dialog-sample-item">
+      <Hds::Dialog open @color={{color}} id="dialog-example-{{color}}" as |D|>
+        <D.Header @icon="info">
+          {{capitalize color}}
+        </D.Header>
+        <D.Body>Dialog content</D.Body>
+        <D.Footer>
+          <form method="dialog" aria-label="Form example {{color}}">
+            <Hds::ButtonSet>
+              <Hds::Button type="submit" @text="Confirm" @color={{if (eq color "critical") "critical"}} />
+              <Hds::Button type="submit" @text="Cancel" @color="secondary" />
+            </Hds::ButtonSet>
+          </form>
+        </D.Footer>
+      </Hds::Dialog>
+    </div>
+  {{/each}}
+
+  <h4 class="dummy-h4">Size</h4>
+  {{#each @model.SIZES as |size|}}
+    <p class="dummy-paragraph">{{capitalize size}}</p>
+    <br />
+    <div class="dummy-dialog-sample-item">
+      <Hds::Dialog open @size={{size}} id="dialog-example-{{size}}" as |D|>
+        <D.Header @icon="info">
+          {{capitalize size}}
+        </D.Header>
+        <D.Body>Dialog content</D.Body>
+        <D.Footer>
+          <form method="dialog" aria-label="Form example {{size}}">
+            <Hds::ButtonSet>
+              <Hds::Button type="submit" @text="Confirm" />
+              <Hds::Button type="submit" @text="Cancel" @color="secondary" />
+            </Hds::ButtonSet>
+          </form>
+        </D.Footer>
+      </Hds::Dialog>
+    </div>
+  {{/each}}
+
+  {{! This below is just an example of invocation, to get started }}
+  <h4 class="dummy-h4">Invocation</h4>
+  <Hds::ButtonSet>
+    <Hds::Button
```

Comment:
This is probably a common problem when disabling page content, but there's a small layout jump when the dialog is invoked due to the scroll bar being removed. Not sure if it's an issue or if it's something that can even be solved, more just a note.

Reply:
Ah, I guess this happens if you set your scrollbars to be always shown in macOS. Good spot – I'll try and think about a workaround. This is caused by setting overflow to hidden on the main body element, not part of the component, but part of our showcase styling.